--- setup.py.old	2016-02-15 15:13:24.407400900 +0100
+++ setup.py	2016-02-15 15:16:20.011658600 +0100
@@ -11,6 +11,7 @@
 import re
 import sys
 from distutils.core import setup, Extension
+from setuptools import setup
 
 define_macros={}
 
@@ -72,6 +73,9 @@
 except:
     pass
 
+if sys.version_info >= (3,):
+    kwargs['use_2to3'] = True
+
 setup(
     name = 'setproctitle',
     description = 'A Python module to customize the process title',
