--- setup.py.old	2016-02-15 15:42:23.842254400 +0100
+++ setup.py	2016-02-15 15:46:45.447388500 +0100
@@ -30,6 +30,13 @@
     
 #from numpy.distutils.system_info import system_info, NotFoundError, dict_append, so_ext
 from numpy.distutils.core import setup
+try:
+    from distutils.command.build_py import build_py_2to3
+except ImportError:
+    pass
+COMMANDS = {}
+if sys.version_info[0] >= 3:
+    COMMANDS['build_py'] = build_py_2to3
 
 DOC_FILES = []
 
@@ -85,5 +92,6 @@
               'Intended Audience :: Developers',
               'Intended Audience :: Science/Research',
               'License :: OSI Approved :: BSD License',
-              'Topic :: Scientific/Engineering']
+              'Topic :: Scientific/Engineering'],
+        cmdclass=COMMANDS,
     )
